primefaces/src/main/java/org/primefaces/validate/FileValidator.java

```diff
@@ -60,51 +63,119 @@ public enum PropertyKeys {
+            throw new IllegalArgumentException("Argument of type '" + value.getClass().getName() + "' not supported");
+        }
+        String accept = resolveAccept(componentAccept);
+        String allowTypes = resolveAllowTypes(componentAccept, component);
+        if (singleFile != null) {
+            validateUploadedFile(context, singleFile, accept, allowTypes);
+        }
+        else if (multipleFiles != null) {
+            validateUploadedFiles(context, multipleFiles, accept, allowTypes);
+        }
+    }
+
+    /**
+     * Resolves the accept value for validation.
+     * If content-type validation is enabled, returns either the validator's allowMediaTypes
+     * or the component's accept attribute.
+     * @param componentAccept Raw accept attribute from the component
+     * @return the accept value to use for validation, or null if content type validation is disabled
+     */
+    private String resolveAccept(String componentAccept) {
+        if (Boolean.TRUE.equals(getContentType())) {
+            if (LangUtils.isNotBlank(getAllowMediaTypes())) {
+                return getAllowMediaTypes();
+            }
+            return componentAccept;
+        }
+        return null;
+    }
+
+    /**
+     * Resolves the allowed file types pattern for validation.
+     * Returns the validator's allowTypes if configured. If allowTypes is not configured
+     * and content-type validation is enabled, extracts file extensions from the component's
+     * accept attribute and converts them to a regex pattern.
+     * @param componentAccept Raw accept attribute from the component
+     * @param component The UI component being validated
+     * @return the file types pattern to use for validation, or null if none found
+     */
+    private String resolveAllowTypes(String componentAccept, UIComponent component) {
+        String allowTypes = getAllowTypes();
+        if (LangUtils.isBlank(allowTypes) && Boolean.TRUE.equals(getContentType())) {
+            String extractedAllowTypes = extractAllowTypes(componentAccept);
+            if (LangUtils.isNotBlank(extractedAllowTypes)) {
+                allowTypes = extractedAllowTypes;
+            }
+            else {
+                LOGGER.log(Level.WARNING, "Attribute allowTypes is missing in p:validateFile. ClientId: " + component.getClientId());
+            }
         }
+        return allowTypes;
+    }
+
+    /**
+     * Extracts file extensions from an HTML accept attribute and converts them to a JavaScript-compatible
+     * regular expression pattern. Only processes dot-prefixed extensions (e.g., ".pdf", ".doc") and
+     * ignores MIME types (e.g., "image/*", "text/plain").
+     * @param componentAccept Raw accept attribute from the component (e.g., ".pdf,.doc,image/*")
+     * @return a regex pattern matching the file extensions (e.g., "/.*\.(pdf|doc)/"), or null if no valid extensions found
+     */
+    private static String extractAllowTypes(String componentAccept) {
```

Comment:
can you please add some units tests?
you could also move it to the FileuploadUtils if necessary for better accessibility.